packages/account/src/Components/poi/idv-document-submit/idv-document-submit.tsx

```diff
@@ -40,178 +39,209 @@ import { API_ERROR_CODES } from 'Constants/api-error-codes';
+        const validateFields = (values: TIdvDocumentSubmitForm) => {
+            const errors: FormikErrors<Record<keyof TIdvDocumentSubmitForm, string>> = {};
+            const { document_type, document_number, document_additional } = values;
+
+            if (document_type.id === IDV_NOT_APPLICABLE_OPTION.id) {
+                return errors;
             }
-            errors.document_number = isDocumentNumberValid(document_number, document_type);
-        }
-        const required_fields = ['first_name', 'last_name', 'date_of_birth'];
-        const validateValues = validate(errors, values);
-        validateValues(val => val, required_fields, localize('This field is required'));
 
-        if (values.first_name) {
-            errors.first_name = validateName(values.first_name);
-        }
-        if (values.last_name) {
-            errors.last_name = validateName(values.last_name);
-        }
+            const needs_additional_document = !!document_type.additional;
 
-        if (!values.confirmation_checkbox) {
-            errors.confirmation_checkbox = 'error';
-        }
+            errors.document_type = isDocumentTypeValid(document_type);
+            if (!shouldHideHelperImage(document_type?.id)) {
+                if (needs_additional_document) {
+                    errors.document_additional = isAdditionalDocumentValid(document_type, document_additional);
+                }
+                errors.document_number = isDocumentNumberValid(document_number, document_type);
+            }
+            const required_fields = ['first_name', 'last_name', 'date_of_birth'];
+            const validateValues = validate(errors, values);
+            validateValues(val => val, required_fields, localize('This field is required'));
 
-        return removeEmptyPropertiesFromObject(errors);
-    };
-
-    const submitHandler = async (
-        values: TIdvDocumentSubmitForm,
-        {
-            setSubmitting,
-            setStatus,
-            status,
-        }: FormikHelpers<TIdvDocumentSubmitForm> & FormikState<TIdvDocumentSubmitForm>
-    ) => {
-        setSubmitting(true);
-
-        const request = makeSettingsRequest(values, changeable_fields);
-
-        const data = await WS.setSettings(request);
-
-        if (data?.error) {
-            const response_error =
-                data.error?.code === API_ERROR_CODES.DUPLICATE_ACCOUNT
-                    ? DUPLICATE_ACCOUNT_ERROR_MESSAGE
-                    : GENERIC_ERROR_MESSAGE;
-            setStatus({ ...status, error_message: response_error });
-            setSubmitting(false);
-            return;
-        }
-        const get_settings = await WS.authorized.storage.getSettings();
-        if (get_settings?.error) {
-            setStatus({ ...status, error_message: get_settings?.error?.message ?? GENERIC_ERROR_MESSAGE });
-            setSubmitting(false);
-            return;
-        }
+            if (values.first_name) {
```

Comment:
should we add values?.first_name

Reply:
it is not necessary, as these fields are defined with initial values